From cd348d0a913720513882ccc5e79bb4bab7ec0e06 Mon Sep 17 00:00:00 2001
From: Fabian Greffrath <fabian@greffrath.com>
Date: Sun, 8 Sep 2019 20:58:54 +0200
Subject: [PATCH 1/9] menu: use Pillow instead of ImageMagick for level preview
 generation

This significantly reduces the dependeny chain. Also, one less call to
os.spawnlp() means a smaller attack surface.
---
 menu/README.md             |  2 +-
 menu/jumpnbump_menu.py.pre | 11 +++++++----
 2 files changed, 8 insertions(+), 5 deletions(-)

diff --git a/menu/README.md b/menu/README.md
index 32544c1..834cd92 100644
--- a/menu/README.md
+++ b/menu/README.md
@@ -15,7 +15,7 @@ Makefile to initialize some packaging-specific constants.
 - Jump 'n Bump
 - Python 3
 - PyGObject
-- ImageMagick
+- Pillow
 
 ## History
 
diff --git a/menu/jumpnbump_menu.py.pre b/menu/jumpnbump_menu.py.pre
index a7dfc2f..bfd8f90 100644
--- a/menu/jumpnbump_menu.py.pre
+++ b/menu/jumpnbump_menu.py.pre
@@ -28,6 +28,8 @@ import gi
 gi.require_version('Gtk', '3.0')
 from gi.repository import Gtk, Gdk, GObject
 
+from PIL import Image
+
 RESOURCE_DIR = '%%DATADIR%%/jumpnbump'
 BINARY_DIR = '%%BINDIR%%'
 UNPACK_BIN = '%%UNPACKBIN%%'
@@ -114,10 +116,11 @@ def level_changed(widget):
         os.chdir(unpackdir)
         os.spawnlp(os.P_WAIT, UNPACK_BIN,
                    'unpack', choosen_level)
-        os.spawnlp(os.P_WAIT, 'convert', 'convert', '-scale',
-                   '50%', 'level.pcx', 'level_scaled.pcx')
-        os.spawnlp(os.P_WAIT, 'convert', 'convert',
-                   'level_scaled.pcx', 'level.png')
+        im = Image.open('level.pcx')
+        w, h = im.size
+        im.thumbnail([w/2, h/2], Image.ANTIALIAS)
+        im.save('level.png')
+        im.close()
         image.set_from_file('level.png')
     except Exception as err:
         print(err)
-- 
2.33.0

